code/modules/vehicle/tg_vehicles/scooter.dm

```diff
@@ -0,0 +1,342 @@
+	else
+		var/backdir = reverse_direction(dir)
+		step(src, backdir)
+		rider.spin(4, 1)
+
+///Moves the vehicle forward and if it lands on a table, repeats
+/obj/tgvehicle/scooter/skateboard/proc/grind()
+	if(!has_buckled_mobs())
+		grinding = FALSE
+		icon_state = "[initial(icon_state)]"
+		return
+	var/mob/living/skater = buckled_mobs[1]
+	var/old_pass = skater.pass_flags //Re-do this, so railings don't fuck with the grinder
+	var/old_v_pass = pass_flags
+	skater.pass_flags |= PASSTABLE | PASSFENCE
+	pass_flags |= PASSTABLE | PASSFENCE
+	step(src, dir)
+	skater.pass_flags = old_pass
+	pass_flags = old_v_pass
+	var/piping_hot = FALSE
+	for(var/obj/machinery/atmospherics/pipe/P in loc.contents)
+		if(P.invisibility == 0 && (loc.layer == 1.7 || P.layer == 2.47))
+			piping_hot = TRUE
+			break
+	if(!(locate(/obj/structure/table) in loc.contents) && !(locate(/obj/structure/railing) in loc.contents) && !piping_hot)
+		grinding = FALSE
+		icon_state = "[initial(icon_state)]"
+		return
+	skater.adjustStaminaLoss(instability*0.3)
+	if(skater.getStaminaLoss() >= 100)
+		playsound(src, 'sound/effects/bang.ogg', 20, TRUE)
+		unbuckle_mob(skater)
+		var/atom/throw_target = get_edge_target_turf(src, pick(NORTH, SOUTH, EAST, WEST))
+		skater.throw_at(throw_target, 2, 2)
+		visible_message("<span class='danger'>[skater] loses [skater.p_their()] footing and slams on the ground!</span>")
+		skater.Weaken(4 SECONDS)
+		grinding = FALSE
+		icon_state = "[initial(icon_state)]"
+		return
+	playsound(src, 'sound/effects/skateboard_roll.ogg', 50, TRUE)
+	var/turf/location = get_turf(src)
+
+	if(location)
+		if(prob(33))
+			location.hotspot_expose(1000,1000)
+			sparks.start() //the most radical way to start plasma fires
+	for(var/mob/living/carbon/victim in location)
+		if(victim.body_position == LYING_DOWN)
+			playsound(location, 'sound/items/trayhit2.ogg', 40)
+			victim.apply_damage(damage = 25, damagetype = BRUTE, def_zone = list("head", "chest", "l_arm", "r_arm", "l_leg", "r_leg"))
+			victim.Weaken(1.5 SECONDS)
+			skater.adjustStaminaLoss(instability)
+			victim.visible_message("<span class='danger'>[victim] straight up gets grinded into the ground by [skater]'s [src]! Radical!</span>")
+	addtimer(CALLBACK(src, PROC_REF(grind)), 1)
+
+/obj/tgvehicle/scooter/skateboard/MouseDrop(atom/over_object)
+	. = ..()
+	var/mob/living/carbon/skater = usr
+	if(!istype(skater))
+		return
+	if(over_object == skater)
+		pick_up_board(skater)
+
+/obj/tgvehicle/scooter/skateboard/proc/pick_up_board(mob/living/carbon/skater)
+	if(skater.incapacitated() || !Adjacent(skater))
+		return
+	if(has_buckled_mobs())
+		to_chat(skater, "<span class='warning'>You can't lift this up when somebody's on it.</span>")
+		return
+	skater.put_in_hands(new board_item_type(get_turf(skater)))
+	qdel(src)
+
+/obj/tgvehicle/scooter/skateboard/pro
+	name = "skateboard"
+	desc = "An EightO brand professional skateboard. Looks a lot more stable than the average board."
+	icon_state = "skateboard2"
```

Comment:
Better rename it to "skateboard_pro" or something other than just 2.

Reply:
done